scripts/serial_link.py

```diff
@@ -382,6 +382,30 @@ def format_log_json_entry(t0, item):
   data = item.to_json_dict()
   return {'delta': delta, 'timestamp': timestamp, 'data': data}
 
+class Watchdog:
+  def __init__(self, timeout, alarm):
+    self.timeout = timeout
+    self.alarm = alarm
+    self.timer = None
+
+  def __call__(self, *args):
+    self.call()
+
+  def call(self):
+    if self.timer:
+      self.timer.cancel()
+    self.timer = threading.Timer(self.timeout, self.alarm)
+    self.timer.start()
+
+  def alarm(self):
+    import thread
+    thread.interrupt_main()
+
+def default_watchdog():
+  print "ERROR: Watchdog expired!"
```

Comment:
nit: may want to use stderr instead: `import os; os.stderr.write('ERROR: Watchdog expired!`).